kasa/tapo/tapodevice.py

```diff
@@ -0,0 +1,157 @@
+            "usage": self._usage,
+            "time": self._time,
+        }
+
+        self._update_state_information()
+
+        _LOGGER.debug("Got an update: %s", self._data)
+
+    def _update_state_information(self):
+        self._state_information = {
+            "is_hw_v2": self._info.hw_ver == "2.0",
+            "overheated": self._info.overheated,
+            "signal_level": self._info.signal_level,
+            "auto_off_status": self._info.auto_off_status,
+            "auto_off_remain_time": self._info.auto_off_remain_time,
+            "SSID": base64.b64decode(self._info.ssid).decode(),
+        }
+
+    @property
+    def sys_info(self) -> Dict[str, Any]:
+        """Returns the device info."""
+        return self._info
+
+    @property
+    def model(self) -> str:
+        """Returns the device model."""
+        return self._info.model
+
+    @property
+    def alias(self) -> str:
+        """Returns the device alias or nickname."""
+        return base64.b64decode(self._info.nickname).decode()
+
+    @property
+    def time(self) -> datetime:
+        """Return the time."""
+        return datetime.fromtimestamp(
+            self._time.timestamp, tz=timezone(timedelta(minutes=self._time.time_diff))
+        )
+
+    @property
+    def timezone(self) -> Dict:
+        """Return the timezone and time_difference."""
+        return {"timezone": timezone(timedelta(minutes=self._time.time_diff))}
+
+    @property
+    def hw_info(self) -> Dict:
+        """Return hardware info for the device."""
+        return {
+            "sw_ver": self._info.fw_ver,
+            "hw_ver": self._info.hw_ver,
+            "mac": self._info.mac,
+            "type": self._info.type,
+            "hwId": self._info.device_id,
+            "dev_name": self.alias,
+            "oemId": self._info.oem_id,
+        }
+
+    @property
+    def location(self) -> Dict:
+        """Return the device location."""
+        loc = {
+            "latitude": self._info.latitude / 10_000,
+            "longitude": self._info.longitude / 10_000,
+        }
+        return loc
+
+    @property
+    def rssi(self) -> Optional[int]:
+        """Return the rssi."""
+        return self._info.rssi
+
+    @property
+    def mac(self) -> str:
+        """Return the mac formatted with colons."""
+        return self._info.mac.replace("-", ":")
+
+    @property
+    def device_id(self) -> str:
+        """Return the device id."""
+        return self._info.device_id
+
+    @property
+    def internal_state(self) -> Any:
+        """Return all the internal state data."""
+        return self._data
+
+    async def _query_helper(
+        self, target: str, cmd: str, arg: Optional[Dict] = None, child_ids=None
+    ) -> Any:
+        res = await self.protocol.query({cmd: arg})
+
+        return res
+
+    @property
+    def state_information(self) -> Dict[str, Any]:
+        """Return the key state information."""
+        return self._state_information
```

Comment:
I'm not sure if it's worth caching/storing the info inside a separate variable, but rather just return it here?

Reply:
Done